### ISSUE TITLE
Got `fatal error: concurrent map writes` when make run.

### ISSUE THREAD

**Issue by wangcanfengxs** [goharbor/harbor-operator#202]
log:
![image](https://user-images.githubusercontent.com/18596457/99947800-7af1e080-2db3-11eb-96ad-8e4026de6e75.png)

**Comment by steven-zou** [comment]
It seems the webhook registration does not support the concurrent register approach.

```go
// Register marks the given webhook as being served at the given path.
// It panics if two hooks are registered on the same path.
func (s *Server) Register(path string, hook http.Handler) {
	s.defaultingOnce.Do(s.setDefaults)
	_, found := s.webhooks[path]
	if found {
		panic(fmt.Errorf("can't register duplicate path: %v", path))
	}
	// TODO(directxman12): call setfields if we've already started the server
	s.webhooks[path] = hook
	s.WebhookMux.Handle(path, instrumentedHook(path, hook))
	log.Info("registering webhook", "path", path)
}
```

panic here `s.webhooks[path] = hook`

the `webhooks` is just a common map:

```go
// webhooks keep track of all registered webhooks for dependency injection,
	// and to provide better panic messages on duplicate webhook registration.
	webhooks map[string]http.Handler
```

**Comment by steven-zou** [comment]
Create an issue to the controller-runtime lib, check https://github.com/kubernetes-sigs/controller-runtime/issues/1285

**Comment by steven-zou** [comment]
They fixed it in https://github.com/kubernetes-sigs/controller-runtime/pull/1225 and fix contained in V0.6.4
